### DISCUSSION TITLE
Capturing step name to send it to Report Portal

### DISCUSSION THREAD

**Discussion by RodrigoBarretosTR** [badeball/cypress-cucumber-preprocessor#989] [Q&A] [answered]
Hello, there!

I'm using:
- Node 16.13.2
- Cypress 12.9.0
- cypress-cucumber-preprocessor 4.3.1
- @reportportal/agent-js-cypress 5.0.3

I need a way to trigger a function to capture the step name and send it to Report Portal.
But, to make this question not to specific I just want to know if is there a way to do the part where I can trigger a function to capture the step name in the version I'm using or any newer lib version.

I did research on the lib code and the issues and found out the I can read the the step name from `window.testState`. But I'm still struggling to find a way to capture this on the right time (every time a step started). Is there any lib event or other way to do that?

_If is there any missing info, please let me know to make my question more clear as possible._

**Comment by badeball** [comment, marked as answer]
> But I'm still struggling to find a way to capture this on the right time (every time a step started). Is there any lib event or other way to do that?

No, the request can be tracked here: https://github.com/badeball/cypress-cucumber-preprocessor/issues/847

**Reply by RodrigoBarretosTR** [reply]
Oh, ok!
Thanks for the reply
